cloudigrade/analyzer/tasks.py

```diff
@@ -118,7 +132,58 @@ def _parse_log_for_ec2_events(log):
     return dict(instances)
 
 
-def _is_on_off_event(record, valid_events):
+def _parse_log_for_ami_tag_events(log):
+    """
+    Parse S3 log for AMI tag create/delete events.
+
+    Args:
+        log (str): The JSON string contents of the log file.
+
+    Returns:
+        None: Images are updated if needed
+
+    """
+    log = json.loads(log)
+
+    for record in log.get('Records', []):
+        if not _is_valid_event(record, ec2_ami_tag_event_list):
+            continue
+
+        add_openshift_tag = record.get('eventName') == CREATE_TAG
+        ami_list = [ami.get('resourceId') for ami in record.get(
+            'requestParameters', {})
+            .get('resourcesSet', {})
+            .get('items', []) if ami.get('resourceId', '').startswith('ami-')]
+
+        tag_list = [tag for tag in record.get(
+            'requestParameters', {})
+            .get('tagSet', {})
+            .get('items', []) if tag.get('key', '') == AWS_OPENSHIFT_TAG]
+
+        if ami_list and tag_list:
+            openshift_tag = ImageTag.objects.filter(
+                description=OPENSHIFT_MODEL_TAG).first()
+            for ami_id in ami_list:
+                ami = AwsMachineImage.objects.filter(ec2_ami_id=ami_id).first()
+                if ami:
+                    if add_openshift_tag:
+                        logger.info(
+                            _('Adding openshift tag to AMI {}').format(
+                                ami_id))
+                        ami.tags.add(openshift_tag)
+                    else:
+                        logger.info(_(
+                            'Removing openshift tag from AMI {}').format(
+                                ami_id))
+                        ami.tags.remove(openshift_tag)
+                    ami.save()
+                else:
+                    logger.info(
+                        'Tag create/delete event referenced AMI {}, '
+                        'but no AMI with this ID is known to cloudigrade.')
+
+
+def _is_valid_event(record, valid_events):
```

Comment:
Was there a reason for this name change? The old name was pretty descriptive, but the new one is a little confusing as it doesn't seem to necessarily be validating if its valid, since it only checks a small subset (on/off events), not all events.

Reply:
Yes. I used it to filter the tag create/delete events too.

Reply:
Yes. I used it to filter the tag create/delete events too.

Reply:
Given it took a list of valid events and I needed the exact same functionality, it made sense to rename it and reuse it for my events too.